kie-server-parent/kie-server-services/kie-server-services-task-assigning/kie-server-services-task-assigning-planning/src/main/java/org/kie/server/services/taskassigning/planning/TaskAssigningConstants.java

```diff
@@ -91,6 +91,32 @@ private TaskAssigningConstants() {
      */
     public static final String TASK_ASSIGNING_USERS_SYNC_INTERVAL = "org.kie.server.taskAssigning.usersSyncInterval";
 
+    /**
+     * Property for configuring the amount of time to execute LS phase before sending any plan to the kie-server with
+     * the jBPM runtime. Allows an early improvement of the solution. This value should typically be low, e.g. 500
+     * milliseconds, etc.
+     *
+     * The formats accepted are based on the ISO-8601 duration format {@code PnDTnHnMn.nS} with days considered to
+     * be exactly 24 hours.
+     * <p>
+     * e.g.
+     * "PT0.500S"  configures 500 milliseconds.
+     */
+    public static final String TASK_ASSIGNING_FIRST_LEVEL_LOCAL_SEARCH_TIME = "org.kie.server.taskAssigning.firstLevelLocalSearchTime";
```

Comment:
yes..., was not direct to find a good name for this configuration parameters (an explanation will go in the documentation in all cases) , good thing about this current names is that the term "LocalSearch", that in the end is what the configuration is about, somehow links to the OptaPlanner LS. Will think a bit

Reply:
That's a question: does the user, in this case probably a system administrator, who is configuring the task assigning, need to understand it runs some local search under the hood?

Should the name of the parameter reflect its observable effect or internal implementation? I would argue rather the former, wdyt?

Reply:
well, in some way, yes, but unfortunately the configuration of all the task assigning integration has many aspects that are very closely to OptaPlanner, and it's not neccesary a bad thing. e.g.
we could eventually configure things like
org.kie.server.taskAssigning.solver.configResource
org.kie.server.taskAssigning.solver.moveThreadCount
org.kie.server.taskAssigning.solver.moveThreadBufferSize
org.kie.server.taskAssigning.solver.threadFactoryClass

and I think it's a good thing, since it's links you to OptaPlanner that is the in the end what we'll always have behind the scene. Additionally it helps to have a quick link with OptaPlanner documentation, etc.

This two new parametes are a kind of "in the middle"